libnmstate/varlink/__init__.py

```diff
@@ -0,0 +1,257 @@
+            },
+        )
+
+
+class NmstateConflictError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateConflictError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateLibnmError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateLibnmError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateVerificationError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateVerificationError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateNotImplementedError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateNotImplementedError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateInternalError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateInternalError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateDependencyError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateDependencyError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class ServiceRequestHandler(varlink.RequestHandler):
+    service = service
+
+
+@service.interface("io.nmstate")
+class NmstateService:
+    def Show(self, include_status_data):
+        """
+        Reports the state data on the system
+        """
+        show_kwargs = {}
+        if include_status_data is not None:
+            show_kwargs["include_status_data"] = include_status_data
+        try:
+            configured_state = libnmstate.show(**show_kwargs)
+            state = {"state": configured_state}
+            return state
+        except libnmstate.error.NmstateValueError as exception:
+            raise NmstateValueError(str(exception))
+
+    def Apply(self, state, verify_change, commit, rollback_timeout):
+        """
+        Apply desired state declared in json format
```

Comment:
`state` is a dictionary not json.

Reply:
But using through varlink it is defined as json string